src/Panel/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx

```diff
@@ -86,318 +77,254 @@ export class TimeLayerSliderPanel extends React.Component<TimeLayerSliderPanelPr
-      return true;
-    }
-    if (nextState.endDate !== endDate) {
-      return true;
-    }
-    if (!(_isEqual(nextProps.timeAwareLayers, timeAwareLayers))) {
-      return true;
-    }
-
-    return !(_isEqual(nextProps.tooltips, tooltips));
-  }
+    autoPlaySpeedOptions = [0.5, 1, 2, 5, 10, 100, 300],
+    initStartDate = moment(moment.now()),
+    initEndDate = moment(moment.now()).add(1, 'days')
+  }) => {
+    const [currentValue, setCurrentValue] = useState<Moment>(value);
+    const [playbackSpeed, setPlaybackSpeed] = useState<
+      number | PlaybackSpeedType
+    >(1);
+    const [autoPlayActive, setAutoPlayActive] = useState(false);
+    const [startDate, setStartDate] = useState<Moment>(initStartDate);
+    const [endDate, setEndDate] = useState<Moment>(initEndDate);
+
+    const wmsTimeLayersRef = useRef<TimeLayerAwareConfig[]>([]);
+    const intervalRef = useRef<number | undefined>(1000);
+    const prevPropsRef = useRef<TimeLayerSliderPanelProps>();
+
+    const wrapTimeSlider = useCallback(() => {
+      const wmsTimeLayers: TimeLayerAwareConfig[] = [];
+      timeAwareLayers.forEach((l: any) => {
+        if (l.get('type') === 'WMSTime') {
+          wmsTimeLayers.push({ layer: l });
+        }
+      });
+      wmsTimeLayersRef.current = wmsTimeLayers;
+    }, [timeAwareLayers]);
+
+    const autoPlay = useCallback(() => {
+      setAutoPlayActive(prevState => !prevState);
+    }, []);
+
+    const onTimeChanged = useCallback(
+      (val: string | [string, string]) => {
+        const newTime = moment(val);
+        setCurrentValue(newTime);
+        if (onChange) {
+          onChange(newTime);
+        }
+        wmsTimeHandler(val);
+      },
+      [onChange]
+    );
 
-  /**
-   * Wraps the TimeSlider component in timeLayerAware.
-   */
-  wrapTimeSlider = () => {
-    this._wmsTimeLayers = [];
-    this.props.timeAwareLayers!.forEach((l: any) => {
-      if (l.get('type') === 'WMSTime') {
-        this._wmsTimeLayers.push({
-          layer: l
-        });
-      }
-    });
+    const updateDataRange = useCallback(([start, end]: [Moment, Moment]) => {
+      setStartDate(start);
+      setEndDate(end);
+    }, []);
+
+    const setSliderToNow = useCallback(() => {
+      const now = moment().milliseconds(0);
+      setCurrentValue(now);
+      setEndDate(now);
+      wmsTimeHandler(now);
+    }, []);
+
+    const onPlaybackSpeedChange = useCallback(
+      (val: number | PlaybackSpeedType) => {
+        setPlaybackSpeed(val);
+      },
+      []
+    );
 
-    // make sure an initial value is set
-    this.wmsTimeHandler(this.state.value);
-  };
-
-  /**
-   * Updates slider time range depending on chosen layer set.
-   */
-  findRangeForLayers = () => {
-    const {
-      timeAwareLayers
-    } = this.props;
-    const {
-      startDate,
-      endDate,
-    } = this.state;
-
-    if (timeAwareLayers.length === 0) {
-      return;
-    }
+    const wmsTimeHandler = (val?: any) => {
```

Comment:
you might be able to use the type `moment.Moment` here instead of any

Reply:
Changed the typing here, as well.